site/src/pages/docs/advanced-spring-boot-integration.mdx

```diff
@@ -0,0 +1,421 @@
+Also, you can add custom configuration of the service using <type://MetricCollectingServiceConfigurator> bean.
+
+
+```java filename=ArmeriaConfiguration.java
+@Bean
+public MeterIdPrefixFunction meterIdPrefixFunction() {
+    return MeterIdPrefixFunction.ofDefault("my.armeria.service");
+}
+
+@Bean
+public MetricCollectingServiceConfigurator metricCollectingServiceConfigurator() {
+    return metricCollectingServiceBuilder -> metricCollectingServiceBuilder
+            .successFunction((context, log) -> {
+                final int statusCode = log.responseHeaders().status().code();
+                // Treat a 404 response as a success
+                return statusCode >= 200 && statusCode < 400 || statusCode == 404;
+            });
+}
+```
+
+For remind, you must not configure above settings using <type://ServerBuilder> directly through
+<type://ArmeriaServerConfigurator> or `Consumer<ServerBuilder>`. It will be overrided by automatic configuration of Armeria.
+
+### Actuator Service
+
+If you are using [Spring Boot Actuator](https://docs.spring.io/spring-boot/docs/current/actuator-api/htmlsingle/),
+you can serve this at `armeria.internal-services.port` and you can serve the internal services at `managment.server.port`.
+
+In order to use the actuator service, you need to add the following dependencies first.
+
+<RequiredDependencies
+  dependencies={[
+    {
+      groupId: 'com.linecorp.armeria',
+      artifactId: 'armeria-spring-boot3-actuator-starter',
+    }
+  ]}
+/>
+
+Include the service id `actuator` to `armeria-internal-services.include`.
+
+```yaml application.yml
+armeria:
+  internal-services:
+    include: actuator
+```
+
+## Other Bean Configuration
+
+We will explain about some additional bean configurations that would be helpful to know.
+
+### <type://ArmeriaServerConfigurator> vs `Consumer<ServerBuilder>`
+
+You can use both <type://ArmeriaServerConfigurator> and `Consumer<ServerBuilder>` to configure the server using <type://ServerBuilder>.
+Actually those two are same which are just function of `ServerBuilder -> void`. The most important difference is the order
+in which they are applied to the server. Armeria first configures all <type://ArmeriaServerConfigurator> beans
+and then applies all `Consumer<ServerBuilder>` beans.
+
+If you have multiple <type://ArmeriaServerConfigurator>s or `Consumer<ServerBuilder>`s, you can set the order by
+[@Order](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/annotation/Order.html) annotation.
+One thing to note is that the default order of <type://ArmeriaServerConfigurator> is `0(zero)`
+not the [`Ordered.LOWEST_PRECEDENCE`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/Ordered.html#LOWEST_PRECEDENCE).
+
+### <type://DependencyInjector>
+
+Armeria can manually inject dependency using <type://DependencyInjector>. You can find the example at [1.17.0 release notes](/release-notes/1.17.0).
```

Comment:
What do you think about explaining the default out of the box behavior first? e.g. by default, with spring boot integration enabled, Armeria uses Spring BeanFactory for injecting dependencies. You can change this behavior by ...

Reply:
Thanks! I agree with your opinion. However, It seems that what you mentioned is not the default behavior based on what I know. Armeria uses Spring BeanFactory for injecting dependencies when `armeria.enable-auto-injection=true`. Therefore, I explained dependency injection using Spring in the later part.